IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -422,7 +420,19 @@ - (MSIDAccountCacheItem *)accountWithKey:(MSIDCacheKey *)key
                                              context:(id<MSIDRequestContext>)context
                                                error:(NSError **)error
 {
-    return [self cacheItemsWithKey:key serializer:serializer cacheItemClass:[MSIDAccountCacheItem class] context:context error:error];
+    /*
+     Sync in memory cache with persistent cache at the time of look up.
+     */
+    MSIDMacCredentialStorageItem *savedStorageItem = [self storageItemWithKey:key serializer:serializer context:context error:error];
+    MSIDMacCredentialStorageItem *storageItem = key.isShared ? self.sharedStorageItem : self.appStorageItem;
```

Comment:
would account ever be not shared?

Reply:
Moved this logic of reading and merging to a common api which will be called by saveToken, saveAccount, saveMetadata, e.t.c. This check will be required now.